### PR TITLE
Expand docs on the side in the suggest widget

### PR DESCRIPTION
**Pull request by ramya-rao-a**
First draft for #18582

Features in the PR:
- If either `suggestion.details` or `suggestion.documentation` exist, they will appear on the side of the suggest list. If no docs but details match then label, then the docs section is not shown
- To use page up and down in the docs, press `ctrl+space` and then page up and down as usual
-  A logic based on cursor position and viewport length is in place to decide when to swap the list and the docs. Not fool proof yet. Just a starting point

Pending:
- [x] More testing on corner scenarios where the list and docs need to be swapped
- [x] Show docs below when editor window is too small
- [ ] A visual cue for when focus is on the docs section vs the list
- [ ] Test accessibility and screen reader

Update as of 5/7: Logic to decide when to swap list and docs, where to position the suggest widget and when to re-size the docs is finalized here: https://github.com/Microsoft/vscode/blob/ramyar/docs-on-the-side/src/vs/editor/contrib/suggest/browser/suggestWidget.ts#L817

### CODE REVIEW

**Comment by joaomoreno** [comment]
I'll file individual issues for the problems I find.

File: src/vs/editor/contrib/suggest/browser/media/suggest.css [hunk 1]
```diff
@@ -23,7 +23,13 @@
 
 .monaco-editor .suggest-widget > .tree {
 	height: 100%;
-	width: 100%;
+	width: 40%;
+	float: left;
```
Comment (joaomoreno) [inline src/vs/editor/contrib/suggest/browser/media/suggest.css]: Careful with floats. If all you have is two divs, you should use the usual `inline-block` display to lay them out side by side.

**Review by joaomoreno — COMMENTED** [review]
Code looks OK. I filed a few issues. Nothing blocking in my opinion, just a lot of tweaking. I'll leave the merge decision up to you.
